src/win/fs.c

```diff
@@ -2627,8 +2627,56 @@ static void fs__statfs(uv_fs_t* req) {
                              &bytes_per_sector,
                              &free_clusters,
                              &total_clusters)) {
-    SET_REQ_WIN32_ERROR(req, GetLastError());
-    return;
+    DWORD err;
+
+    err = GetLastError();
+    if (err == ERROR_DIRECTORY) {
+      WCHAR* pathw;
+      WCHAR* fpart;
+      size_t len;
+      DWORD ret;
+
+      len = MAX_PATH + 1;
+      pathw = uv__malloc(len * sizeof(*pathw));
+retry:
+      ret = GetFullPathNameW(req->file.pathw,
```

Comment:
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew#remarks warns that `GetFullPathNameW` is not thread safe

Reply:
For thread safe issues, I have created a new issue(#2696). This is probably the case if we pass a relative path to all functions that take a file name as an argument, not just this PR.